### ISSUE TITLE
Slack/Gitter integration

### ISSUE THREAD

**Issue by sedubois** [OpenCollective/api#186]
Cfr. https://trello.com/c/b7Kz2Lgl/385-slack-gitter-integration. We want to integrate with Slack or Gitter. Similar to our own integration, so someone can know right away when an activity happens.

Both Gitter and Slack allow receiving messages after authentication with oauth. Gitter also has webhooks. Slack webhooks without oauth apparently can't be used for other teams than our own.

Using oauth should provide a better user experience as user doesn't need to copy-paste a URL between two different pages, but requires more work.

Gitter also offers the possibility through a custom integration (https://github.com/gitterHQ/services) to let user configure which events he subscribes to (e.g. user created, group transaction added, etc). But that apparently only supports webhooks, not oauth flow.

The Gitter API allows to create (POST) the chatroom to post messages to. "If the room doesn't exist but the supplied URI represents a GitHub Org or GitHub Repo the user is an admin of, the room will be created automatically and the user added." In the case of Slack, AFAIK the channel is chosen during authentication flow.

In both cases, some backend and UI changes would be needed to support the authentication flow and store the credentials.

Other Gitter pros/cons:
- Pros:
  - a demo app is provided which seems to explain well what is required: https://github.com/gitterHQ/gitter-demo-app
- Cons:
  - API is beta
  - can't provide an icon with individual messages. But I suppose that's OK, if all OpenCollective messages go to a specific chatroom anyway
  - only allows showing Open Collective as an "app" on Gitter website when using webhooks, not oauth (currently)

Other Slack pros/cons: 
- Pros:
  - using this API opens the way to later add features like triggering opencollective commands straight from Slack, and realtime communication with a "bot user"
- Cons: 
  - can't apparently let user configure what specific events they receive
  - overall documentation seems more complicated

In summary if we're after something quick, Gitter webhooks would do the trick, but if we want to see long term a Slack oauth integration might be better.

What are your thoughts? Gitter vs. Slack, webhook vs. oauth? etc.
@arnaudbenard @asood123 @xdamman

**Comment by sedubois** [comment]
First Slack version cfr. https://github.com/OpenCollective/api/pull/199
Gitter: https://github.com/OpenCollective/api/pull/191
Closing.
